python/az/aro/azext_aro/_validators.py

```diff
@@ -283,3 +289,74 @@ def validate_load_balancer_managed_outbound_ip_count(namespace):
     if namespace.load_balancer_managed_outbound_ip_count < minimum_managed_outbound_ips or namespace.load_balancer_managed_outbound_ip_count > maximum_managed_outbound_ips:  # pylint: disable=line-too-long
         error_msg = f"--load-balancer-managed-outbound-ip-count must be between {minimum_managed_outbound_ips} and {maximum_managed_outbound_ips} (inclusive)."  # pylint: disable=line-too-long
         raise InvalidArgumentValueError(error_msg)
+
+
+def validate_enable_managed_identity(namespace):
+    if not namespace.enable_managed_identity:
+        return
+
+    if namespace.client_id is not None:
+        raise InvalidArgumentValueError('Must not specify --client-id when --enable-managed-identity is True')
+
+    if namespace.client_secret is not None:
+        raise InvalidArgumentValueError('Must not specify --client-secret when --enable-managed-identity is True')
+
+    if namespace.version is None or not re.match(r'^[4-9]{1}\.[0-9]{1,2}\.[0-9]{1,2}$', namespace.version):
+        raise InvalidArgumentValueError('Enabling managed identity requires --version >= 4.14.z')
+
+    _, versionY, _ = namespace.version.split('.', 2)
+    if int(versionY) < 14:
+        raise InvalidArgumentValueError('Enabling managed identity requires --version >= 4.14.z')
+
+    if not namespace.platform_workload_identities:
+        raise RequiredArgumentMissingError('Enabling managed identity requires platform workload identities to be provided')  # pylint: disable=line-too-long
+
+    if not namespace.mi_user_assigned:
+        raise RequiredArgumentMissingError('Enabling managed identity requires cluster identity to be provided')
+
+
+def validate_platform_workload_identities(cmd, namespace):
+    if namespace.platform_workload_identities is None:
+        return
+
+    if not namespace.enable_managed_identity:
+        raise RequiredArgumentMissingError('Must set --enable-managed-identity when providing platform workload identities')  # pylint: disable=line-too-long
+
+    for identity in namespace.platform_workload_identities:
+        if not is_valid_resource_id(identity.resource_id):
+            identity.resource_id = identity_name_to_resource_id(
+                cmd, namespace, identity.resource_id)
+
+        if not identity_resource_id_is_valid(identity.resource_id):
+            raise InvalidArgumentValueError(f"Resource {identity.resource_id} used for platform workload identity {identity.name} is not a valid userAssignedIdentity")  # pylint: disable=line-too-long
+
+
+def validate_cluster_identity(cmd, namespace):
+    if namespace.mi_user_assigned is None:
+        return
+
+    if not namespace.enable_managed_identity:
+        raise RequiredArgumentMissingError('Must set --enable-managed-identity when providing a cluster identity')  # pylint: disable=line-too-long
+
+    if not is_valid_resource_id(namespace.mi_user_assigned):
+        namespace.mi_user_assigned = identity_name_to_resource_id(
```

Comment:
It seems it does more than validation.
I think it would be better that we stick to validation in validate functions, and if we need formalization, do somewhere else, otherwise it will be confusing.

btw is it safe to assume that `identity.mi_user_assigned` is a MI name when it is not valid?

Reply:
Same response as https://github.com/Azure/ARO-RP/pull/3624#discussion_r1657599264